pkg/policy/repository_test.go

```diff
@@ -2118,3 +2118,88 @@ func (ds *PolicyTestSuite) TestremoveIdentityFromRuleCaches(c *C) {
 
 	c.Assert(addedRule.metadata.IdentitySelected, checker.DeepEquals, map[identity.NumericIdentity]bool{})
 }
+
+func (ds *PolicyTestSuite) TestIterate(c *C) {
+	repo := NewPolicyRepository(nil, nil, nil)
+	repo.selectorCache = testSelectorCache
+
+	numWithEgress := 0
+	countEgressRules := func(r *api.Rule) {
+		if len(r.Egress) > 0 {
+			numWithEgress++
+		}
+	}
+	repo.Mutex.Lock()
+	repo.IterateLocked(countEgressRules)
```

Comment:
who's accessing this concurrently? Looks like you create it above. Is `NewPolicyRepository` spawning some go-routines?